addons/stock/models/stock_production_lot.py

```diff
@@ -10,17 +10,17 @@ class ProductionLot(models.Model):
     _description = 'Lot/Serial'
 
     name = fields.Char(
-        'Serial Number', default=lambda self: self.env['ir.sequence'].next_by_code('stock.lot.serial'),
-        required=True, help="Unique Serial Number")
-    ref = fields.Char('Internal Reference', help="Internal reference number in case it differs from the manufacturer's serial number")
+        'Lot/Serial Number', default=lambda self: self.env['ir.sequence'].next_by_code('stock.lot.serial'),
+        required=True, help="Unique Lot/Serial Number")
+    ref = fields.Char('Internal Reference', help="Internal reference number in case it differs from the manufacturer's lot/serial number")
     product_id = fields.Many2one(
         'product.product', 'Product',
         domain=[('type', 'in', ['product', 'consu'])], required=True)
     quant_ids = fields.One2many('stock.quant', 'lot_id', 'Quants', readonly=True)
     create_date = fields.Datetime('Creation Date')
 
     _sql_constraints = [
-        ('name_ref_uniq', 'unique (name, product_id)', 'The combination of serial number and product must be unique !'),
+        ('name_ref_uniq', 'unique (name, product_id)', 'The combination of lot/serial number and product must be unique !'),
```

Comment:
this constraint only apply to the serial number. A lot might have combination with several products.